src/Yoti/Util/Profile/AnchorConverter.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Yoti\Util\Profile;
+
+use Traversable;
+use phpseclib\File\ASN1;
+use phpseclib\File\X509;
+use Attrpubapi_v1\Anchor;
+use Yoti\Entity\Anchor as YotiAnchor;
+
+class AnchorConverter
+{
+    public function __construct()
+    {
+        $this->ASN1 = new ASN1();
+        $this->X509 = new X509();
+    }
+
+    /**
+     * Convert Protobuf Anchors to a map of Yoti Anchors
+     * And merge the result into the reference array
+     *
+     * @param Anchor $anchor
+     * @param $yotiAnchorsMap
+     */
+    public function convertToYotiAnchors(Anchor $anchor, &$yotiAnchorsMap)
```

Comment:
This could be a little simpler...

_convertToYotiAnchors_ - this method works on one Anchor at a time, so the method name should be singular

It would be a bit cleaner if this method simply returned the new _Anchor_, rather than adding it to the map.  Let the _AnchorListConverter_ manage the collection.
So then we would have _AnchorConverter_, which just converts _Anchors_ one at a time, and separately we'd have _AnchorListConverter_ which converts a list of Anchors into the Map.

Reply:
The thing is each protobuf anchor may result to more than one YotiAnchor. As YotiAnchors are produced for each protobuf Anchor extensions.

Reply:
Do you mean that because there's a list of certificates in the AnchorProto there could be multiple Anchors coming out?

It's not necessarily clear from the structure, but we can be confident that one AnchorProto will result in one Anchor.  There are some notes in SDK-397, the ticket for the original implementation.

Reply:
You may have two verifiers on a single attribute. e.g my address has two verifiers

Verifier YOTI_IDENTITY
Verifier YOTI_ADMIN